src/sparseml/transformers/compression/quantization_format.py

```diff
@@ -42,7 +46,28 @@ def infer_quantization_format(
         return quantization_format
 
     if save_compressed:
+        quant_depths = _get_quant_depths(model)
+        if quant_depths == [4]:  # save packed if everything is int4
+            return CompressionFormat.pack_quantized
+
+        # otherwise just quantize to int8
         return CompressionFormat.int_quantized
     else:
         # format will be inferred from config
         return None
+
+
+def _get_quant_depths(model):
```

Comment:
don't we have this func in `compressed-tensors` already?

Reply:
Link?? I didn't think we had anything similar